### PR TITLE
#140: Global export mode not affect for email testing

### PR DESCRIPTION
**Pull request by yogyrton**
https://github.com/RonasIT/laravel-helpers/issues/140

### CODE REVIEW

File: src/Traits/MailsMockTrait.php [hunk 1]
```diff
@@ -63,6 +63,10 @@ protected function assertMailEquals(string $mailableClass, $emailChain, bool $ex
         $emailChain = $this->prepareEmailChain($emailChain);
         $index = 0;
 
+        if (!$exportMode) {
```
Comment (aizlee) [inline src/Traits/MailsMockTrait.php:66]: Need to move `globalMode` checking to the `MailsMockTrait.assertFixture` method.

```
$globalExportMode = $this->globalExportMode ?? false;

if ($exportMode || $globalExportMode) {
    $this->exportContent($mailContent, $expectedMailData['fixture']);
}
```

**Review by aizlee — CHANGES_REQUESTED** [review]

**Review by aizlee — APPROVED** [review]

File: tests/MailsMockTraitTest.php [hunk 2]
```diff
@@ -118,6 +118,19 @@ public function testMailWithExport()
         $this->assertFileExists($this->getFixturePath('test_mail_with_export.html'));
     }
 
+    public function testMailWithGlobalExportMod()
```
Comment (DenTray) [inline tests/MailsMockTraitTest.php:121]:
```suggestion
    public function testMailWithGlobalExportMode()
```

File: tests/MailsMockTraitTest.php [hunk 3]
```diff
@@ -118,6 +118,19 @@ public function testMailWithExport()
...
+    {
+        putenv('FAIL_EXPORT_JSON=false');
+
+        Mail::to('test@mail.com')->queue(new TestMail(['name' => 'John Smith']));
+
+        $this->assertMailEquals(TestMail::class, [
+            $this->mockedMail('test@mail.com', 'test_mail_with_export.html', 'Test Subject'),
```
Comment (DenTray) [inline tests/MailsMockTraitTest.php:128]:
```suggestion
            $this->mockedMail('test@mail.com', 'test_mail_with_global_export.html', 'Test Subject'),
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: tests/MailsMockTraitTest.php [hunk 4]
```diff
@@ -118,6 +118,19 @@ public function testMailWithExport()
...
+    {
+        putenv('FAIL_EXPORT_JSON=false');
+
+        Mail::to('test@mail.com')->queue(new TestMail(['name' => 'John Smith']));
+
+        $this->assertMailEquals(TestMail::class, [
+            $this->mockedMail('test@mail.com', 'test_mail_with_global_export.html', 'Test Subject'),
+        ]);
+
+        $this->assertFileExists($this->getFixturePath('test_mail_with_export.html'));
```
Comment (DenTray) [inline tests/MailsMockTraitTest.php:131]: why you checking `test_mail_with_export.html` file, but asserting content with the `test_mail_with_global_export.html`?

File: tests/fixtures/MailsMockTraitTest/test_mail_with_global_export.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<div>
```
Comment (DenTray) [inline tests/fixtures/MailsMockTraitTest/test_mail_with_global_export.html:1]: you using this fixture to check that globa export mode works fine and create file if it not exists. If you'll commit this file to git - test will always be successful completed even if global export mode will be broken

**Review by DenTray — CHANGES_REQUESTED** [review]

**Pull request closed without merging**